packages/block-editor/src/components/block-switcher/style.scss

```diff
@@ -46,11 +46,13 @@
 }
 
 .components-button.block-editor-block-switcher__no-switcher-icon {
-	width: $block-toolbar-height;
+	display: flex;
+	padding: ($grid-unit-15 * 0.5) $grid-unit-15 !important;
```

Comment:
I know the answer is "specificity", but what does this important need to override? Could we have done with adding some extra classes just around this one, or does the existance of the important vastly simplify the code otherwise?

In general whenever we add an important, we're creating a headace for ourselves in the future. There are occasional cases where it's fine, usually a) when we know this will be refactored in the near future, or b) there's never a case where the rule should be broken.

Reply:
This is the rule I'm overriding with the `!important` here:

```
.block-editor-block-toolbar .components-toolbar-group .components-button.has-icon.has-icon
```

> In general whenever we add an important, we're creating a headace for ourselves in the future. There are occasional cases where it's fine, usually a) when we know this will be refactored in the near future, or b) there's never a case where the rule should be broken.

I really hope for option A here 😅

Reply:
Yep, sounds good. Should we add a comment as to why the important was used?

Reply:
> The `!important` is used to override the selector rule with "specificity hacks".

Something like this?

Reply:
Sure that works. Another version; I like to give as  much context as I can, and conditions for the comment to be removed. For example:

> The !important is used to vastly simplify the overriding of an inherited selector.
> Can be removed if we refactor `.block-editor-block-toolbar .components-toolbar-group .components-button.has-icon.has-icon`

Reply:
🚀